src/main/java/org/jabref/gui/help/VersionWorker.java

```diff
@@ -0,0 +1,94 @@
+ * later).
+ *
+ * If the versions check is executed manually and this is the latest version it will also display a dialog to inform the
+ * user.
+ */
+public class VersionWorker {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(VersionWorker.class);
+
+    /**
+     * The current version of the installed JabRef
+     */
+    private final Version installedVersion;
+
+    /**
+     * The version which was previously ignored by the user
+     */
+    private final Version toBeIgnored;
+    private final DialogService dialogService;
+    private final TaskExecutor taskExecutor;
+
+    public VersionWorker(Version installedVersion, Version toBeIgnored, DialogService dialogService, TaskExecutor taskExecutor) {
+        this.installedVersion = Objects.requireNonNull(installedVersion);
+        this.toBeIgnored = Objects.requireNonNull(toBeIgnored);
+        this.dialogService = Objects.requireNonNull(dialogService);
+        this.taskExecutor = Objects.requireNonNull(taskExecutor);
+    }
+
+    /**
+     * Returns a newer version excluding any non-stable versions, except if the installed one is unstable too. If no
+     * newer version was found, then an empty optional is returned.
+     */
+    private Optional<Version> getNewVersion() throws IOException {
+        List<Version> availableVersions = Version.getAllAvailableVersions();
+        return installedVersion.shouldBeUpdatedTo(availableVersions);
+    }
+
+    /**
+     * @param manualExecution if this versions check is executed automatically (eg. on startup) or manually by the user
+     */
+    public void checkForNewVersionAsync(boolean manualExecution) {
+        BackgroundTask.wrap(this::getNewVersion)
+                      .onSuccess(version -> showUpdateInfo(version, manualExecution))
+                      .onFailure(exception -> showConnectionError(exception, manualExecution))
+                      .executeWith(taskExecutor);
+    }
+
+    /**
+     * Prints the connection problem to the status bar and shows a dialog if it was executed manually
+     */
+    private void showConnectionError(Exception exception, boolean manualExecution) {
+        String couldNotConnect = Localization.lang("Could not connect to the update server.");
+        String tryLater = Localization.lang("Please try again later and/or check your network connection.");
+        if (manualExecution) {
+            dialogService.showErrorDialogAndWait(Localization.lang("Error"), couldNotConnect + "\n" + tryLater, exception);
+        }
+        LOGGER.warn(couldNotConnect + " " + tryLater, exception);
+    }
+
+    /**
+     * Prints up-to-date to the status bar (and shows a dialog it was executed manually) if there is now new version.
+     * Shows a "New Version" Dialog to the user if there is.
+     */
+    private void showUpdateInfo(Optional<Version> newerVersion, boolean manualExecution) {
```

Comment:
This optional parameter is coming from the Background Task?

Reply:
Yes!